[CS2113-W13-2] CLI-nton

NicholasTanYY opened the pull request:
Simplifies life by providing a fast and straightforward way for users to manage tasks, stay informed with news, and check the weather—all within a command-line environment, ensuring a seamless and productive user experience.

cirelesna commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,457 @@
+
+
+## Data Component
+### API: [Data](https://github.com/AY2324S2-CS2113-W13-2/tp/tree/master/src/main/java/data)
+![Data Class Diagram](images/class/Data.png)
```
Could the visual be split up into smaller parts?
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/96415327/1f35598c-e11a-47de-b417-41f28df6b11b)

nikhil-2101 replied:
more diagrams can be included, such as Generic Sequence Diagram.

cirelesna commented on docs/DeveloperGuide.md:
```diff
@@ -20,10 +468,10 @@
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ...             | So that I can ...                                           |
+|---------|----------|---------------------------|-------------------------------------------------------------|
+| v1.0    | new user | see usage instructions    | refer to them when I forget how to use the application      |
+| v2.0    | user     | find a to-do item by name | locate a to-do without having to go through the entire list |
```
Could there be more user stories for v1.0 and v2.0?

cirelesna commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,457 @@
+### Overview: 
+The TaskManager is responsible for handling the creation, reading, updating and deletion [CRUD] of tasks.
+
+### How it works:
+1. When the program starts, a TaskManager instance is created.
+2. Saved tasks are retrieved by interfacing with the Storage class to create required tasks and save them to this 
+instance of TaskManager.  
+3. When the user requests an operation, the main function parses the input and hands it to TaskManager if it is related
+4. The TaskManager will handle all requests relating to tasks, using methods detailed in the following sections.
+   to tasks.
+5. The TaskManager will also create INFO-level logs when making changes to tasks.
+
```
Could a sequence diagram illustrate how it works better?

nikhil-2101 commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,457 @@
+
+The `Main` class is responsible for handling user input and dispatching commands to the appropriate components. It manages the switching between the week and month views based on the user's input.
+
+When the user enters the `month` command, the `Main` class toggles the `inMonthView` flag and calls the `printView` method of the `MonthView` instance, rendering the month view. Similarly, when the user enters the `week` command, the `Main` class sets the `inMonthView` flag to `false` and calls the `printView` method of the `WeekView` instance, rendering the week view.
```
inconsistent formatting when compared to other sections of the developer's guide. Other sections utilised bullet points to convey the necessary information while this section conveyed all of its points in a paragraph.

nikhil-2101 commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,457 @@
 
```
Great job on including very detailed descriptions of each component of your code.

cirelesna commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,457 @@
+## Storage component
+**API** : [Storage.java](https://github.com/AY2324S2-CS2113-W13-2/tp/blob/master/src/main/java/storage/Storage.java)
+
+The 'storage' component:
```
Could a class diagram be used here?

cirelesna reviewed (commented):
DG review

nikhil-2101 commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,457 @@
+        }
+    }
+}
+
```
great job on providing a code snippet to allow us to understand this functionality.

jthibault03 replied:
good usage of code and markdown to illustrate what is happening. Overall quite a clear DG, but maybe more examples of how to use each feature and expected output would be helpful.

daryltay415 commented on docs/images/class/Data.png:
You could remove the attributes in the classes to make it simpler

daryltay415 commented on docs/DeveloperGuide.md:
```diff
@@ -4,9 +4,457 @@
+### Update Manager Method
+
+The `updateManager` method facilitates the updating of tasks. It prompts the user for an updated task description and handles tasks of different types (Todo, Event, Deadline) while ensuring the changes are correctly reflected in the application.
+
```
Maybe you could add a sequence diagram to illustrate it?

clarencepohh commented on docs/DeveloperGuide.md:
Hello, I think your DG review is on the wrong group. Perhaps you are supposed to review T14-2: OmniTravel instead?

jthibault03 replied:
oops, you are correct. I resubmitted the DG review on the right group

clarencepohh committed:
Replace unscrollable code snippets (when in PDF) with images

clarencepohh committed:
Remove duplicate segment

clarencepohh committed:
Add task type and classification summary

clarencepohh committed:
Add table of contents to start of UG

clarencepohh committed:
Remove incorrect notation and add note for users

clarencepohh committed:
Add back notation for date/time entries to conform to Jon's changes

clarencepohh committed:
Merge branch 'master' into Developer-Guide

# Conflicts:
#	docs/diagrams/class/Data.puml

clarencepohh committed:
Fix merge conflict issues

clarencepohh committed:
Add skeleton PPP files and update AboutUs.md

clarencepohh committed:
Add product introduction for README

clarencepohh committed:
Move product introduction

clarencepohh committed:
Add target user profile description

clarencepohh committed:
Add value proposition for DG

clarencepohh committed:
Add NFR and glossary for DG

clarencepohh committed:
Add v1.0 and v2.0 user stories

kyhjonathan committed:
Refactor Exceptions into package

kyhjonathan committed:
Create storage exception class

kyhjonathan committed:
Create checkFileReadable method

kyhjonathan committed:
Refactor to storageFileException

Mahesh1772 committed:
Add displayable icons

Mahesh1772 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

NicholasTanYY committed:
Merge pull request #142 from clarencepohh/UG-branch

Improve UG formatting and fix documentation errors from PE-D

Aak242 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

Mahesh1772 committed:
Add Command and Parsing class, java docs

NicholasTanYY committed:
Merge pull request #146 from clarencepohh/Developer-Guide

DG and README improvements

kyhjonathan committed:
Update Storagefileexceptions

Aak242 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

kyhjonathan committed:
Refactor TaskManager methods into shorter methods

kyhjonathan committed:
Refactor updateManager methods

Mahesh1772 committed:
Correct inconsistencies in parsing class

kyhjonathan committed:
Remove unused imports

kyhjonathan committed:
Ammend checkstyle errors for more than 120 characters in a line

kyhjonathan committed:
Amend checkstyle test errors

Aak242 committed:
Develop add, update task exception handling, implement feature to check for all free/available times in a day

Aak242 committed:
Fixed issues with JUnit tests

kyhjonathan committed:
Add PPP

kyhjonathan committed:
Update PPP

kyhjonathan committed:
Merge branch 'master' into master

kyhjonathan committed:
Merge pull request #148 from kyhjonathan/master

Added exception handling for date parsing and storage and refactoring of taskManager methods

NicholasTanYY committed:
Add overall contributions for PPP

NicholasTanYY committed:
Add rough template for summary of contributions

kyhjonathan committed:
Merge pull request #150 from NicholasTanYY/branch-PPP

Add Nicholas PPP

Mahesh1772 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

# Conflicts:
#	src/main/java/seedu/duke/Main.java

clarencepohh committed:
Merge pull request #147 from Mahesh1772/master

clarencepohh committed:
Update clarencepohh.md

Committing directly to master due to unavailability of a computer, so committing here as I have no IDE/git at the moment. Apologies.

Mahesh1772 committed:
PPP

Mahesh1772 committed:
Error correction

Mahesh1772 committed:
Merge pull request #151 from Mahesh1772/master

PPP

Aak242 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

# Conflicts:
#	src/main/java/data/TaskManager.java
#	src/main/java/seedu/duke/Main.java

Aak242 committed:
Fix JUnit tests for latest merge

Aak242 committed:
Add description of contributions to PPP

clarencepohh committed:
Merge pull request #149 from Aak242/master

Develop add, update task exception handling, implement feature to check for all free/available times in a day

clarencepohh committed:
Merge pull request #152 from Aak242/branch-PPP

Add description of contributions to PPP - Aakash

clarencepohh committed:
Remove unused template file

Aak242 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

kyhjonathan committed:
Add JavaDocs for TaskManager methods

kyhjonathan committed:
Implement more Javadocs for TaskManager

kyhjonathan committed:
Add javadocs for Storage and TaskManager classes

kyhjonathan committed:
Javadocs for commandhandler class

clarencepohh committed:
Move Data segment to be under Design and Implementation

clarencepohh committed:
Add Table of Contents to DG

clarencepohh committed:
Update UG TOC to use internal links

NicholasTanYY committed:
Add unused AvatarUi class

NicholasTanYY committed:
Add unused TerminalController class

NicholasTanYY committed:
Add javadocs for unused code

NicholasTanYY committed:
Add minor javadoc comments for unused code

NicholasTanYY committed:
Fix bug for month command which switches to week view

NicholasTanYY committed:
Fix irregular printing of avatar

clarencepohh committed:
Update formatting on UG and DG

Mahesh1772 committed:
Fix add in Week and MonthView, Add JUnit tests

Mahesh1772 committed:
Fix DateTimeException

NicholasTanYY committed:
Shrink avatar to fit on most screens

NicholasTanYY committed:
Add more logging statements

NicholasTanYY committed:
Add assertions

NicholasTanYY committed:
Add tests for marking tasks as complete

NicholasTanYY committed:
Add tests for marking tasks as not complete

NicholasTanYY committed:
Add test for getting day tasks

NicholasTanYY committed:
Remove unused string constant

NicholasTanYY committed:
Update inputs for text ui tests

NicholasTanYY committed:
Fix line wrapping coding standards

kyhjonathan committed:
Add more JUnit for TaskManager

Mahesh1772 committed:
Fix bugs

kyhjonathan committed:
Remove unused imports

Mahesh1772 committed:
Merge branch 'master' into master

NicholasTanYY committed:
Rename variables for sequence diagram

clarencepohh committed:
Merge pull request #155 from Mahesh1772/master

Fix add in Week and MonthView, Add JUnit tests

clarencepohh committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-W13-2/tp

NicholasTanYY committed:
Merge branch 'master' into branch-final-changes

clarencepohh committed:
Merge pull request #156 from NicholasTanYY/branch-final-changes

Final amendments to project

clarencepohh committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-W13-2/tp

clarencepohh committed:
Merge branch 'master' into Developer-Guide

clarencepohh committed:
Fix testing method names

NicholasTanYY committed:
Update printWeekViewSequenceDiagram

clarencepohh committed:
Add update description only for event test

clarencepohh committed:
Add update description and date/time for events test

clarencepohh committed:
Add date time validation method testing

kyhjonathan committed:
Add in more JUnit tests for exception handlings for wrong time format

clarencepohh committed:
Replace constructor with more accurate constructor

clarencepohh committed:
Add tests for MarkTaskException

kyhjonathan committed:
Add JUnit for checkifValidDate exception

kyhjonathan committed:
Add JUnit for checkIfValidTime exception thrown

kyhjonathan committed:
Add additional JUnits for checkIfDateInCurrentMonth

clarencepohh committed:
Remove unneeded lines

clarencepohh committed:
Add tests for SetPriorityException

clarencepohh committed:
Add new StorageFileExceptionTest class

clarencepohh committed:
Update method to throw exception

clarencepohh committed:
Add StorageFileException tests

clarencepohh committed:
Fix camel case for method names

kyhjonathan committed:
Fix checkstyle test and main errors

kyhjonathan committed:
Merge branch 'master' into master

Mahesh1772 committed:
DG additions

kyhjonathan committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-W13-2/tp

# Conflicts:
#	src/main/java/data/TaskManager.java

kyhjonathan committed:
Merge branch 'master' of https://github.com/kyhjonathan/tp

# Conflicts:
#	src/main/java/data/TaskManager.java

Mahesh1772 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

kyhjonathan committed:
Update exception handling for updating tasks that is out of array range on a valid day

kyhjonathan committed:
Remove indented line

Aak242 committed:
Add Exceptions section in DG, and make pUML class diagram for the same

Aak242 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

kyhjonathan committed:
Refactor unused code

kyhjonathan committed:
Remove unused imports from Event Class

clarencepohh committed:
Update image used for UML

kyhjonathan committed:
Edit user guide for ics documentation

NicholasTanYY committed:
Merge pull request #158 from clarencepohh/final-changes

Add JUnit tests for exception classes

Aak242 committed:
Remove unused functions from TaskManagerException and put them in the appropriate unused folder

kyhjonathan committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-W13-2/tp

# Conflicts:
#	src/test/java/data/TaskManagerExceptionTest.java

clarencepohh committed:
Merge pull request #157 from kyhjonathan/master

Add in Javadocs and JUnit testing

Aak242 committed:
Fix checkstyle

Aak242 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

Aak242 committed:
Delete duplicate TaskManagerExceptionTest

clarencepohh committed:
Merge branch 'master' into final-changes

kyhjonathan committed:
Add in tests for storage functionalities

kyhjonathan committed:
Merge pull request #159 from Aak242/master

Add exception and logging section to DG and move unused code to the appropriate folder

kyhjonathan committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-W13-2/tp

NicholasTanYY committed:
Add UIRenderer class diagram

clarencepohh committed:
Merge branch 'master' into final-changes

NicholasTanYY committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp into branch-final-changes

kyhjonathan committed:
Update UG for my explicit explanation

Mahesh1772 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

# Conflicts:
#	docs/DeveloperGuide.md

kyhjonathan committed:
Change storage txt file name

kyhjonathan committed:
Update Javadocs

clarencepohh committed:
Remove logging section

clarencepohh committed:
Remove error handling section

clarencepohh committed:
Reorder the items in DG to make more sense

clarencepohh committed:
Update PUML and diagram image to be used

clarencepohh committed:
Change visibility for some functions in TaskManager

NicholasTanYY committed:
Add dependency of UiRenderer to TaskManager

NicholasTanYY committed:
Add note to point to TaskManager class UML

Mahesh1772 committed:
DG and UG changes Part 1

Mahesh1772 committed:
UG and DG changes

kyhjonathan committed:
Add architecture diagram

kyhjonathan committed:
Update architucture description in DG

kyhjonathan committed:
Add non-functional requirements

kyhjonathan committed:
Mark exporting ics file as future dev

kyhjonathan committed:
Update DG for name of txt file for storage

kyhjonathan committed:
Update storage component DG

kyhjonathan committed:
Add exception description for error handling in DG

kyhjonathan committed:
Add UML class diagram of storage into DG

kyhjonathan committed:
Implement puml code for sequence diagram of storage methods

kyhjonathan committed:
Add sequence diagram for storage

kyhjonathan committed:
Add sequence diagram for addManager into DG

Mahesh1772 committed:
DG and UG issue fixes

kyhjonathan committed:
Change to screenshots for example in deletion

NicholasTanYY committed:
Merge pull request #189 from clarencepohh/final-changes

Fix DG issues

NicholasTanYY committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp into branch-final-changes

NicholasTanYY committed:
Add FAQ for adding event

Mahesh1772 committed:
Add images

NicholasTanYY committed:
Update command summary in UG for greater clarity

NicholasTanYY committed:
Add free times in the command summary

Aak242 committed:
DG changes

Aak242 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

NicholasTanYY committed:
Add known issues in event management in FAQ

Aak242 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

# Conflicts:
#	docs/DeveloperGuide.md

NicholasTanYY committed:
Add text ui test cases

Aak242 committed:
Fixed checkstyle errors

kyhjonathan committed:
Add screenshot examples for marking UG

kyhjonathan committed:
Add screenshots for UG for priority, mark and quit

kyhjonathan committed:
Add explanation in avoidance of using of dashes and slashes in UG

NicholasTanYY committed:
Update PPP

NicholasTanYY committed:
Add specification to free command in UG

NicholasTanYY committed:
Add reflections in PPP

NicholasTanYY committed:
Update PPP formatting

NicholasTanYY committed:
Change description of UiRenderer in DG

NicholasTanYY committed:
Modify method descriptions in DG

NicholasTanYY committed:
Add DG for method to printTasksInWeek

clarencepohh committed:
Create PUML for Data sequence diagram and its image

NicholasTanYY committed:
Update printHelp method in DG

clarencepohh committed:
Update PPP

clarencepohh committed:
Add PUML into DG for Data sequence diagram

clarencepohh committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-W13-2/tp into final-changes

NicholasTanYY committed:
Update sequence diagram for printing the week's view

clarencepohh committed:
Fix typo

NicholasTanYY committed:
Update sequence diagram in DG

clarencepohh committed:
Update links in PPP

NicholasTanYY committed:
Add Ui class diagram image to DG

NicholasTanYY committed:
Add JUnit tests for AvatarUi

clarencepohh committed:
Add old user story that was missed out

kyhjonathan committed:
Amend JUnit test case names to correct format

kyhjonathan committed:
Amend naming convention of JUnits in StringParserTest

NicholasTanYY committed:
Add JUnit testing for printWeekHeader

kyhjonathan committed:
Amend authorship in TaskManagerExceptionTest and add to DG

Mahesh1772 committed:
Add Contribution and correct image path

kyhjonathan committed:
Edit test cases in StringParserTest

NicholasTanYY committed:
Add more JUnit test cases for UIRenderer

NicholasTanYY committed:
Fix JUnit test cases

Aak242 committed:
Fixed issue with update task not updating on UI

kyhjonathan committed:
Add in more extensive JUnit testing for TaskManagerTest

kyhjonathan committed:
Add more extensive testing for TaskManagerTest

clarencepohh committed:
Remove unneeded whitespace

NicholasTanYY committed:
Merge pull request #194 from clarencepohh/final-changes

Final changes to PPP and DG

NicholasTanYY committed:
Merge branch 'master' into master

NicholasTanYY committed:
Merge pull request #160 from Mahesh1772/master

Some DG changes

kyhjonathan committed:
Update PPP

NicholasTanYY committed:
Update test cases to add line separators

NicholasTanYY committed:
Add lineSeparators to fix window failures

Aak242 committed:
Fix tests and formatting issues

NicholasTanYY committed:
Resolve merge conflicts

NicholasTanYY committed:
Remove test case which fails lineseparator test

kyhjonathan committed:
Merge pull request #196 from NicholasTanYY/final-branch

Final changes

kyhjonathan committed:
Merge branch 'master' into master

kyhjonathan committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

# Conflicts:
#	docs/DeveloperGuide.md

kyhjonathan committed:
Merge branch 'master' of https://github.com/kyhjonathan/tp

# Conflicts:
#	docs/DeveloperGuide.md

kyhjonathan committed:
Handle merge conflicts and DG issues

kyhjonathan committed:
Fix exception handling bug in TaskManager

Aak242 committed:
Add free time description to UserGuide, and change the output images to not show colon for time for the event and deadline update feature

Aak242 committed:
Fix checkstyle test

kyhjonathan committed:
Update QandA UG

Aak242 committed:
Made necessary changes to DG, removed unnecessary classes from the sequence diagram to make it more readable, corrected image path

kyhjonathan committed:
Resolve testing issues

Aak242 committed:
Fix image path

NicholasTanYY committed:
Merge pull request #190 from kyhjonathan/master

Amendments to DG and more test cases

Aak242 committed:
Make change to string parser

kyhjonathan committed:
Amend PPP

kyhjonathan committed:
Change title of PPP

NicholasTanYY committed:
Resolve merge conflicts

NicholasTanYY committed:
Resolve merge conflicts

NicholasTanYY committed:
Merge branch 'Aak242-master'

Aak242 committed:
Update PPP with necessary sections

Mahesh1772 committed:
Merge pull request #197 from kyhjonathan/master

Amend PPP

Mahesh1772 committed:
Merge pull request #198 from NicholasTanYY/master

Fixing changes from aakash' PR

Mahesh1772 committed:
Change PPP

Aak242 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

# Conflicts:
#	docs/DeveloperGuide.md
#	src/main/java/data/TaskManager.java

kyhjonathan committed:
Edit UG mistakes

NicholasTanYY committed:
Merge branch 'master' into master

kyhjonathan committed:
Change formating of DG

NicholasTanYY committed:
Merge pull request #200 from Aak242/master

Add PPP and resolve merge issues

NicholasTanYY committed:
Resolve merge conflicts

clarencepohh committed:
Merge pull request #199 from Mahesh1772/master

Change PPP

clarencepohh committed:
Merge pull request #201 from NicholasTanYY/master

Resolve merge conflicts

kyhjonathan committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113T-W13-2/tp

# Conflicts:
#	docs/DeveloperGuide.md

NicholasTanYY committed:
Merge pull request #202 from kyhjonathan/master

Final PR
